web/packages/design/src/TextArea/TextArea.tsx

```diff
@@ -28,50 +33,114 @@ import {
   HeightProps,
 } from 'styled-system';
 
+import { Theme } from 'design/theme/themes/types';
+
+export type TextAreaSize = 'large' | 'medium' | 'small';
+
 export interface TextAreaProps
   extends ColorProps,
     SpaceProps,
     WidthProps,
     HeightProps {
+  size?: TextAreaSize;
   hasError?: boolean;
   resizable?: boolean;
 
-  // TS: temporary handles ...styles
-  [key: string]: any;
+  // TextArea element attributes
+  autoFocus?: boolean;
+  disabled?: boolean;
+  id?: string;
+  name?: string;
+  readOnly?: boolean;
+  value?: string;
+  defaultValue?: string;
+  placeholder?: string;
+  autoComplete?: HTMLInputAutoCompleteAttribute;
+  spellCheck?: boolean;
+  style?: React.CSSProperties;
+
+  'aria-invalid'?: HTMLAttributes<'textarea'>['aria-invalid'];
+  'aria-describedby'?: HTMLAttributes<'textarea'>['aria-describedby'];
+
+  onChange?: React.ChangeEventHandler<HTMLInputElement>;
+  onKeyPress?: React.KeyboardEventHandler<HTMLInputElement>;
+  onKeyDown?: React.KeyboardEventHandler<HTMLInputElement>;
+  onFocus?: React.FocusEventHandler<HTMLInputElement>;
+  onBlur?: React.FocusEventHandler<HTMLInputElement>;
+  onClick?: React.MouseEventHandler<HTMLInputElement>;
 }
 
-export const TextArea = styled.textarea<TextAreaProps>`
+export const textAreaGeometry: {
+  [s in TextAreaSize]: {
+    height: number;
+    typography: keyof Theme['typography'];
+  };
+} = {
+  large: {
+    height: 96,
+    typography: 'body1',
+  },
+  medium: {
+    height: 84,
+    typography: 'body2',
+  },
+  small: {
+    height: 76,
+    typography: 'body3',
+  },
+};
+
+export const TextArea = forwardRef<HTMLTextAreaElement, TextAreaProps>(
+  ({ size = 'medium', ...rest }, ref) => (
+    <StyledTextArea ref={ref} taSize={size} {...rest} />
+  )
+);
+
+type StyledTextAreaProps = Omit<TextAreaProps, 'size'> & {
+  taSize: TextAreaSize;
+};
+
+const StyledTextArea = styled.textarea<StyledTextAreaProps>`
   appearance: none;
-  border: 1px solid ${props => props.theme.colors.text.muted};
+  border: 1px solid;
+  border-color: ${props =>
+    props.theme.colors.interactive.tonal.neutral[2].background};
   border-radius: 4px;
   box-sizing: border-box;
+  display: block;
   min-height: 50px;
-  height: 80px;
-  font-size: 16px;
-  padding: 16px;
+  height: ${props => textAreaGeometry[props.taSize].height}px;
+  padding: 8px 16px;
   outline: none;
   width: 100%;
+  background-color: transparent;
   color: ${props => props.theme.colors.text.main};
-  background: inherit;
+
+  ${props => props.theme.typography[textAreaGeometry[props.taSize].typography]}
+
+  &:hover {
+    border: 1px solid ${props => props.theme.colors.text.muted};
```

Comment:
Personally, I never liked having a hover state on the inputs :p
I find it distracting and mostly unnecessary because when you hover over a field, the cursor changes, so you know the field is interactive anyway.

Many pages and apps don't style this state, for example GitHub and Slack.

I found this quite old discussion, I think the arguments make sense https://ux.stackexchange.com/a/34436.
This issue is not a dealbreaker for me, but I'm curious about your opinion @roraback!

Reply:
This is an interesting hypothesis, but I can't find any research or case studies around it. Some major design systems (e.g. Material Design) still have hover states for inputs, but I also recognize that a lot of sites and forms don't. I've always assumed the latter is because of the rise of flat design (with fewer clear signifiers of interactivity) and mobile-first design (where the hover state is typically moot), rather than because it is a UX enhancement. And I could just as easily believe the argument that _no_ hover state adds cognitive load for the user to figure out whether clicking on something will do anything. Meanwhile, form fields are probably quite clear when they show up grouped in a form... but they may be less obvious when they appear in isolation, so I'm curious to see if there is any research that dives into whether there is a difference in usability depending on context.

From a consistency standpoint, I like the idea that every interactive element in Teleport has a distinct hover state, even if it is sometimes subtle... but if you can find any research papers or case studies that delve into the topic, I'd love to see them and am happy to reconsider our approach. Meanwhile I'll keep looking...